### DISCUSSION TITLE
Signing commits with ssh keys

### DISCUSSION THREAD

**Discussion by jeremy-rifkin** [community/community#40047] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm trying to sign my commits with an ssh key but my commits are showing up as unverified on github. It's the same ssh key I am using to push to the repository and the fingerprint for the key used to sign does in fact match the fingerprint of the key I uploaded. Why don't the commits show up as verified?

**Comment by DSDmark** [comment]
Hi @jeremy-rifkin,

there is some reason, maybe.

Any of the following is true:- 
- The commit is signed, but the signature could not be verified.
- The commit is not signed, and the committer has enabled vigilant mode.
- The commit is not signed, and an author has enabled vigilant mode.

https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits#about-vigilant-mode

**Comment by airtower-luna** [comment, marked as answer]
When adding an SSH key you can choose whether you want to use it for authentication or signing. Your existing authentication key won't automatically be accepted for signing. I'm not sure if you can re-add the same key to allow signing too, or if you'll have to use a separate key. Generally it's good practice to use separate keys for separate purposes, regardless of what the UI allows.

![grafik](https://user-images.githubusercontent.com/5814919/203929747-9f686089-2f8c-43e2-bdbf-9f6be50ebe18.png)

**Comment by jeremy-rifkin** [comment]
Oh I completely missed that thank you so much @airtower-luna

**Reply by bbrown** [reply]
You have to tell `git` locally to use your SSH key to sign. I suspect there wasn't a help article when you asked it, but [there is now](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-ssh-key).
